src/OrchardCore.Modules/OrchardCore.Queries/Sql/SqlQuerySource.cs

```diff
@@ -46,32 +51,46 @@ public async Task<IQueryResults> ExecuteQueryAsync(Query query, IDictionary<stri
     {
         var metadata = query.As<SqlQueryMetadata>();
 
-        var sqlQueryResults = new SQLQueryResults();
+        var sqlQueryResults = new SQLQueryResults { Items = [] };
 
         var tokenizedQuery = await _liquidTemplateManager.RenderStringAsync(metadata.Template, NullEncoder.Default,
             parameters.Select(x => new KeyValuePair<string, FluidValue>(x.Key, FluidValue.Create(x.Value, _templateOptions))));
 
-        var dialect = _session.Store.Configuration.SqlDialect;
+        var configuration = _session.Store.Configuration;
 
-        if (!SqlParser.TryParse(tokenizedQuery, _session.Store.Configuration.Schema, dialect, _session.Store.Configuration.TablePrefix, parameters, out var rawQuery, out var messages))
+        if (!SqlParser.TryParse(
+                tokenizedQuery,
+                configuration.Schema,
+                configuration.SqlDialect,
+                configuration.TablePrefix,
+                parameters,
+                out var rawQuery,
+                out var messages))
         {
-            sqlQueryResults.Items = [];
-
+            _logger.LogError("Couldn't parse SQL query: {Messages}", string.Join(' ', messages));
             return sqlQueryResults;
         }
 
         await using var connection = _dbConnectionAccessor.CreateConnection();
 
         await connection.OpenAsync();
 
-        if (query.ReturnContentItems)
+        await using var transaction = await connection.BeginTransactionAsync(configuration.IsolationLevel);
+        var queryResults = await connection.QueryAsync(rawQuery, parameters, transaction);
+
+        if (!query.ReturnContentItems)
         {
-            using var transaction = await connection.BeginTransactionAsync(_session.Store.Configuration.IsolationLevel);
-            var queryResult = await connection.QueryAsync(rawQuery, parameters, transaction);
+            sqlQueryResults.Items = queryResults
+                .Select<object, JsonObject>(document => JObject.FromObject(document, _jsonSerializerOptions))
+                .ToList();
```

Comment:
```suggestion
                .ToArray();
```

Reply:
In most cases `ToList` is considered faster e.g. see [here](https://stackoverflow.com/a/60103725/417315).

Reply:
At allocation cost https://github.com/OrchardCMS/OrchardCore/discussions/15240#discussioncomment-8348547